### ISSUE TITLE
Move early stopping to pytorch ignite engine

### ISSUE THREAD

**Issue by fabiansinz** [sinzlab/neuralpredictors#30]
I feel the early stopping gets overloaded with functionality:
* keep track of best model
* log scores
* early stop
* learning rate schedule

Since that doesn't even cover all our use cases, we should move to something more flexible. I feel that PyTorch Ignite might do that (https://pytorch.org/ignite/). Their central concept is that of an `Engine` that loops through data and does things. What it does is controlled by user-provided functions. In addition, steps like learning rate decrease or early stopping can be handled by provided or custom events triggered at the start of an epoch, at every batch, when the training is completed, and so on. 

The above cases could be handled as follows:
* Best model: After every epoch, the validation score is tested and a best model is stored if it improved compared to before
* Log scores: Before every epoch, a number of scorers are evaluated.
* Early stop: Early Stopping already has a handler in Ignite. 
* Learning rate scheduler: Can be handled with batch or epoch events
* Adapting model components such as fixing the sampling of the Gaussian readout can be handled by custom events such as `FIRST_LR_DECREASE` or so. 

Opinions? Does anyone want to program a toy example?
